#! /bin/sh
patch -p1 -f $* < $0
exit $?

From 7b978c1ce250eeedae897a9a6a2b105b7ebd7f77 Mon Sep 17 00:00:00 2001
From: Markus Montkowski <markus.montkowski@gnupg.com>
Date: Fri, 25 Apr 2025 12:41:03 +0200
Subject: [PATCH] Fix bug introduced fixing T6646 on April 9

* ribbon-callback.cpp: get_mail_from_control fix accesing an
empty vector
--

Calling front() on a vector where empty() is true is
undefined. Code rearanged to prevent that.

Fixes-commit: a95d340e478f8fb9529b49b0518a0fafa4d00bfe
GnuPG-bug-id: 6646
---
 src/ribbon-callbacks.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/ribbon-callbacks.cpp b/src/ribbon-callbacks.cpp
index 06af440..32d5c87 100644
--- a/src/ribbon-callbacks.cpp
+++ b/src/ribbon-callbacks.cpp
@@ -556,13 +556,15 @@ get_mail_from_control (LPDISPATCH ctrl, bool *none_selected)
   if (ret==NULL)
   {
     auto retV = Mail::searchMailsByUUID(uid);
-    if (retV.empty())
+    if (!retV.empty())
+      {
+        ret = retV.front();
+      }
+    else
       {
         log_error ("%s:%s: Failed to find mail %p in map.",
                   SRCNAME, __func__, mailitem);
       }
-    ret = retV.front();
   }
   xfree (uid);

--
2.39.2
